### DISCUSSION TITLE
Is it possible to reference a posts own frontmatter?

### DISCUSSION THREAD

**Discussion by jonohewitt** [mdx-js/mdx#1280] [Q&A] [answered]
E.g

```
---
title: Example
variableArray: ["A", "B", "C"]
---

Text

<CustomComponent propName={frontmatter.variableArray} />
```

Thanks!

**Comment by deadcoder0904** [comment]
I don't think so. What's the use-case for this?

If you want to programmatically add `CustomComponent` using `frontmatter.variableArray`, then you can read the `frontMatter` & insert it dynamically in the MDX using some regex but still failing to understand the use-case for this 🤔

**Reply by jonohewitt** [reply]
Thanks for looking into this! I was trying to render data from variableArray within the MDX body while also making the data available for querying elsewhere in my project via graphql without repeating it. As the answer below says, using an export works well.

**Comment by ChristopherBiscardi** [comment, marked as answer]
you would be able to do this if you used exports instead of frontmatter. MDX itself doesn't officially support frontmatter. frontmatter is an addon some implementations provide for compatibility with their existing ecosystems (such as gatsby-remark)

**Reply by jonohewitt** [reply]
Thanks!
